### DISCUSSION TITLE
How to apply exclude_none recursively on nested models in Pydantic v2?

### DISCUSSION THREAD

**Discussion by Paillat-dev** [pydantic/pydantic#10817] [Question] [answered]
I have nested models where I want `exclude_none` to apply recursively to all nested models when dumping. Currently, it seems that `exclude_none` only applies to the top-level model.

Minimal example to illustrate:
```python
from pydantic import BaseModel
from typing import Optional

class Inner(BaseModel):
    required_field: str
    optional_field: Optional[str] = None

class Outer(BaseModel):
    inner: Inner
    top_optional: Optional[str] = None

# Test case
inner = Inner(required_field="test")  # optional_field is None
outer = Outer(inner=inner)  # both optional fields are None

dumped = outer.model_dump_json()
print(dumped)
# Current output:
# {"inner": {"required_field": "test", "optional_field": null}, "top_optional": null}

# Desired output:
# {"inner": {"required_field": "test"}}
```

Is there a way to make `exclude_none` apply recursively to all nested models without having to set it on each model class? This would be particularly useful when working with APIs that treat omitted fields differently from explicit `null` values.

**Comment by off6atomic** [comment, marked as answer]
bro you didn't even put `exclude_none` anywhere in the code; if you actually do `exclude_none=True`, your code would work recursively:
```python
dumped = outer.model_dump_json(exclude_none=True)
print(dumped)
# {"inner":{"required_field":"test"}}
```

**Reply by Paillat-dev** [reply]
Tbh I have no idea what my issue was but I probably closed it because of what you just mentioned. I have no idea what the context of my post was
